### PR TITLE
Fairscale import updates

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Follow up to #14675 
Suggested by @carmocca 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

cc @borda @justusschock @awaelchli @rohitgr7 @tchaton @carmocca @akihironitta

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]

File: src/pytorch_lightning/overrides/fairscale.py [hunk 1]
```diff
@@ -14,20 +14,16 @@
...
+from lightning_lite.strategies.fairscale import _FAIRSCALE_AVAILABLE
 from pytorch_lightning.overrides.base import (
     _LightningModuleWrapperBase,
     _LightningPrecisionModuleWrapperBase,
     unwrap_lightning_module,
 )
-from pytorch_lightning.utilities.imports import _IS_WINDOWS
 from pytorch_lightning.utilities.rank_zero import rank_zero_deprecation
 
-_FAIRSCALE_AVAILABLE = not _IS_WINDOWS and module_available("fairscale.nn")
```
Comment (Borda) [inline src/pytorch_lightning/overrides/fairscale.py:28]: well this os for meta package creation

Comment (awaelchli) [reply src/pytorch_lightning/overrides/fairscale.py:28]: Ok, I'm not sure actually I understand what's going on. I don't see how fairscale has anything to do with the meta package creation, but indeed the build fails so this is strange.

Comment (Borda) [reply src/pytorch_lightning/overrides/fairscale.py:28]: The challenge here is the whole class is declared inside the `if` condition...
Inside the meta, we replace add func and class declaration with imports so as `_FAIRSCALE_AVAILABLE` is imported in the PL module, it is not imported meta module, so it is missing in the module... the solution is to:
1. keep `_FAIRSCALE_AVAILABLE` declaration even if it is duplicate
2. declare the class outside `if` conditions which would need some more refactor

Comment (awaelchli) [reply src/pytorch_lightning/overrides/fairscale.py:28]: Thank you, now it makes more sense! 

> declare the class outside if conditions which would need some more refactor

I will try this first, seems like the easiest solution.

**Review by rohitgr7 — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by otaj — APPROVED** [review]

**Pull request merged**